src/msg/node.h

```diff
@@ -25,9 +25,13 @@ class CRNode {
     using job_t           = JobRunner::job_t;
     using strand_ptr_t    = std::shared_ptr<JobRunnerStrand>;
 
-    explicit CRNode() : CRNode("noname") {}
+    explicit CRNode() : CRNode("noname", nullptr) {}
 
-    explicit CRNode(std::string name) : name_(std::move(name)) {}
+    explicit CRNode(std::string name) : CRNode(std::move(name), nullptr) {}
+
+    explicit CRNode(std::shared_ptr<JobRunner> runner) : CRNode("noname", std::move(runner)) {}
```

Comment:
weak_ptr?

Reply:
We will save it as a weak_ptr. Passing in a shared_ptr will
- make sure we have a valid runner when constructing nodes
- If we need to access the runner in ctor, we don't need to check if the wp is valid.